src/deploy/extensions/secrets.ts

```diff
@@ -0,0 +1,232 @@
+        secretParam.param
+      }. ` +
+        `projects/${projectId}/secrets/${secretName} is managed by a different extension instance (${
+          secretInfo.labels[secretUtils.SECRET_LABEL]
+        }), so reusing it here can lead to unexpected behavior. ` +
+        "Please choose a different name for this secret, and rerun this command."
+    );
+  }
+}
+
+async function handleSecretParamForUpdate(
+  secretParam: Param,
+  i: InstanceSpec,
+  prevValue: string,
+  nonInteractive: boolean
+) {
+  const providedValue = i.params[secretParam.param];
+  if (!providedValue) {
+    return;
+  }
+  const [, projectId, , secretName, , version] = providedValue.split("/");
+  if (!projectId || !secretName || !version) {
+    throw new FirebaseError(
+      `${clc.bold(i.instanceId)}: Found '${providedValue}' for secret param ${
+        secretParam.param
+      }, but expected a secret version.`
+    );
+  }
+  // Don't allow changing secrets, only changing versions
+  const [, prevProjectId, , prevSecretName] = prevValue.split("/");
+  if (prevProjectId !== projectId || prevSecretName !== secretName) {
+    throw new FirebaseError(
+      `${clc.bold(i.instanceId)}: Found '${providedValue}' for secret param ${
+        secretParam.param
+      }, ` +
+        `but this instance was previously using a different secret projects/${prevProjectId}/secrets/${prevSecretName}.\n` +
+        `Changing secrets is not supported. If you want to change the value of this secret, ` +
+        `use a new version of projects/${prevProjectId}/secrets/${prevSecretName}.` +
+        `You can create a new version at ${secretManager.secretManagerConsoleUri(projectId)}`
+    );
+  }
+  const secretInfo = await getSecretInfo(projectId, secretName, version);
+  if (!secretInfo.secret) {
+    await promptForCreateSecret({
+      projectId,
+      secretName,
+      instanceId: i.instanceId,
+      secretParam,
+      nonInteractive,
+    });
+  } else if (!secretInfo.secretVersion) {
+    throw new FirebaseError(
+      `${clc.bold(i.instanceId)}: Found '${providedValue}' for secret param ${
+        secretParam.param
+      }. ` +
+        `projects/${projectId}/secrets/${secretName} exists, but version ${version} does not. ` +
+        `See more information about this secret at ${secretManager.secretManagerConsoleUri(
+          projectId
+        )}`
+    );
+  }
+}
+
+async function getSecretInfo(
+  projectId: string,
+  secretName: string,
+  version: string
+): Promise<{
+  secret?: secretManager.Secret;
+  secretVersion?: secretManager.SecretVersion;
+  labels: Record<string, string>;
+}> {
+  const secretInfo: any = {
+    labels: {},
+  };
+  try {
+    secretInfo.secret = await secretManager.getSecret(projectId, secretName);
+    secretInfo.secretVersion = await secretManager.getSecretVersion(projectId, secretName, version);
+    secretInfo.labels = await secretManager.getSecretLabels(projectId, secretName);
```

Comment:
optional: getSecretLabel is identical to getSecret two lines above, it just only returns labels and the getSecret doesn't include labels... we should really combine those two....

Reply:
I mean, it's my own fault... for implementing it this way, I just didn't need those two in the same place.... :)

Reply:
Haha, I noticed this and made a mental note to fix it later, then forgot to fix it later. I'll dedupe this logic